fms/utils/serialization.py

```diff
@@ -0,0 +1,132 @@
+from collections import ChainMap, OrderedDict
+import os
+from pathlib import Path
+from typing import Any, List
+import torch
+
+__adapters = {}
+
+
+def register_adapter(architecture: str, source: str, adapter: type):
```

Comment:
should we have an abstract class for adapter here? I think it would make sense to have one as it enforces the contract as stated in the doc

Reply:
actually that type hint is probably wrong. initially I was thining an adapter might be a class (and it can be... sort of...) ... but really it's just a callable from dict to dict that may or may not be a constructor.

Reply:
updated the type hint